src/conversions/bytes.rs

```diff
@@ -0,0 +1,161 @@
+#![cfg(feature = "bytes")]
+
+//! Conversions to and from [bytes](https://docs.rs/bytes/latest/bytes/)'s [`Bytes`] and
+//! [`BytesMut`] types.
+//!
+//! This is useful for efficiently converting Python's `bytes` and `bytearray` types efficiently.
+//!
+//! # Setup
+//!
+//! To use this feature, add in your **`Cargo.toml`**:
+//!
+//! ```toml
+//! [dependencies]
+#![doc = concat!("pyo3 = { version = \"", env!("CARGO_PKG_VERSION"),  "\", features = [\"bytes\"] }")]
+//! bytes = "1.10"
+//!
+//! Note that you must use compatible versions of bytes and PyO3.
+//!
+//! # Example
+//!
+//! Rust code to create functions which return `Bytes` or take `Bytes` as arguements:
+//!
+//! ```rust,no_run
+//! use pyo3::prelude::*;
+//!
+//! #[pyfunction]
+//! fn get_message_bytes() -> Bytes {
+//!     Bytes::from(b"Hello Python!".to_vec())
+//! }
```

Comment:
It would be more efficient to create a `PyBytes` object directly here. This is an example of why I'm not a fan of the Rust to Python conversion.

Reply:
Also `Vec<u8>` and `&[u8]` which are idiomatic bytes types convert with no additional cost.

That said, I think here if we used `Bytes::from_static` it is also no overhead compared to `&[u8]`.

```suggestion
//! #[pyfunction]
//! fn get_message_bytes() -> Bytes {
//!     Bytes::from_static(b"Hello Python!")
//! }
```